pkg/sql/lease.go

```diff
@@ -563,6 +579,15 @@ type tableState struct {
 		// If set, leases are released from the store as soon as their
 		// refcount drops to 0, as opposed to waiting until they expire.
 		dropped bool
+		// asyncRenewalInProgress indicates an asynchronous routine to renew the
+		// table descriptor lease has begun. The flag is required because when
+		// acquisition is done asynchronously, we can have a race condition where
+		// another routine attempts to also renew the lease before the async routine
+		// acquires t.mu.
+		asyncRenewalInProgress bool
```

Comment:
i think you can eliminate the need for this and use acquiring != nil as long as you restructure acquireNodeLease() to take in an asynchronous boolean which asks it to acquire the lease asynchronously and then structure it such that the channel is closed by the function that does the lease acquisition